euphonic/readers/castep.py

```diff
@@ -194,19 +207,28 @@ def read_phonon_data(
             # Sometimes there are more than n_qpts q-points in the file
             # due to LO-TO splitting
             if idx < len(qpts):
+                qpt_ids[idx] = qpt_id
                 qpts[idx] = qpt
                 freqs[idx] = qfreq
                 weights[idx] = qweight
                 if read_eigenvectors:
                     eigenvecs[idx] = qeigenvec
             else:
+                qpt_ids = np.concatenate((qpt_ids, [qpt_id]))
                 qpts = np.concatenate((qpts, [qpt]))
                 freqs = np.concatenate((freqs, [qfreq]))
                 weights = np.concatenate((weights, [qweight]))
                 if read_eigenvectors:
                     eigenvecs = np.concatenate((eigenvecs, [qeigenvec]))
             idx += 1
 
+    # Multiple qpts with same CASTEP q-pt index: correct weights
+    if average_repeat_points:
+        id_counts = {qpt_id: np.count_nonzero(qpt_ids == qpt_id)
+                     for qpt_id in repeated_qpt_ids}
+        for qpt_id, count in id_counts.items():
+            weights[qpt_ids == qpt_id] /= count
```

Comment:
Ah, I think I see what you mean. I was trying to avoid was running counts, which are a bit tricky and harder to troubleshoot than these collections of meaningful information. But we could have the best of both with something like:

```python
repeated_qp_ids = defaultdict(set)
while True:
...
    if average_repeat_points and qpt_id == previous_qpt_id:
        repeated_qpt_ids[qpt_id].update({idx - 1, idx})
...
```
Then we have some collections of indices to play with.

Reply:
Looks and works great! The only weird thing is that the `quartz_nosplit.phonon` file has 6 gamma points in a row, labelled `5, 5, 6, 6, and 7, 7`. The algorithm handles these as expected, and it looks like it would handle a case with more duplicate points too e.g. `5, 5, 5, 5, 6, 6`. But I don't think we'd ever see that in the wild (..right?) so I'm not sure it would be worth testing that.

Reply:
We shouldn't have to worry about the "identifiable as Gamma" thing now that `is_gamma` isn't even imported any more.

I'll have a go at the second one.

Reply:
Done!